internal/templates/antiaffinity/template.go

```diff
@@ -0,0 +1,71 @@
+package antiaffinity
+
+import (
+	"fmt"
+
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/extract"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/antiaffinity/internal/params"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Anti affinity not specified",
+		Key:         "anti-affinity",
+		Description: "Flag objects with multiple replicas but anti affinity not specified in the pod template spec",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(_ *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				replicas, found := extract.Replicas(object.K8sObject)
+				if !found {
+					return nil
+				}
+				if replicas > 1 {
+					podTemplateSpec, hasPods := extract.PodTemplateSpec(object.K8sObject)
+					if !hasPods {
+						return nil
+					}
+					if podTemplateSpec.Spec.Affinity != nil && podTemplateSpec.Spec.Affinity.PodAntiAffinity != nil {
+						preferredAffinity := podTemplateSpec.Spec.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
+						requiredAffinity := podTemplateSpec.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+						for _, preferred := range preferredAffinity {
+							labelSelector, err := metaV1.LabelSelectorAsSelector(preferred.PodAffinityTerm.LabelSelector)
+							if err != nil {
+								return []diagnostic.Diagnostic{{
```

Comment:
Instead of returning, let's just ignore these and continue. We need to add separate checks to validate objects in general, but I would prefer to keep this check focused on this.